src/js/SimpleRPCClient.js

```diff
@@ -0,0 +1,75 @@
+export default class SimpleRPCClient{
+  constructor(backendUrl){
+    this.backendUrl = backendUrl
+    this.eventListenersMap = {}
+    this.socket = null
+    this.sendQueue = []
+  }
+
+  connect(){
+    return new Promise((resolve, reject) => {
+      if(!this.backendUrl){
+        reject();
+        return
+      }
+
+      this.socket = new WebSocket(this.backendUrl)
+      this.socket.onopen = (evt) => {
+        resolve();
+      }
+      this.socket.onerror = (evt) => {
+        console.log('Failed to connect to Simple RPC Client')
+        reject();
+      }
+      this.socket.onclose = (evt) => {
+        console.log("Closed Simple RPC Client connection")
+      }
+      this.socket.onmessage = (evt) => {
+        let event = JSON.parse(evt.data)
+        let event_name = event.method;
+        if(event.success != undefined && event_name in this.eventListenersMap){
+          let params = event.params
+          this.eventListenersMap[event_name](event.success, params)
+        }
+      }
+    });
+  }
+
+  disconnect(){
+    if (this.socket){
+      this.socket.close()
+      this.socket = null
+    }
+  }
+
+  sendJSONMessage(json_msg){
+    let msg = JSON.stringify(json_msg)
+    this.sendQueue.push(msg)
+    this.triggerSend()
+  }
+
+  triggerSend(){
+    var that = this;
+    setTimeout(() => {that.onSend()}, 100)
```

Comment:
Why not just call `onSend` here?

Reply:
The function is also used here https://github.com/smartdevicelink/generic_hmi/pull/239/files#diff-1e5a0d12a111c4e05c96cbae67033ea2R63 to call onSend in case `this.socket.readyState != this.socket.OPEN`. The `100` delay is just to give it time to resolve that state